docs/src/api/class-locator.md

```diff
@@ -0,0 +1,945 @@
+to resolve and return its value.
+
+Examples:
+
+```js
+const elements = page.locator('div');
+const divCounts = await elements.evaluateAll((divs, min) => divs.length >= min, 10);
+```
+
+```java
+Locator elements = page.locator("div");
+boolean divCounts = (boolean) elements.evaluateAll("(divs, min) => divs.length >= min", 10);
+```
+
+```python async
+elements = page.locator("div")
+div_counts = await elements("(divs, min) => divs.length >= min", 10)
+```
+
+```python sync
+elements = page.locator("div")
+div_counts = elements("(divs, min) => divs.length >= min", 10)
+```
+
+```csharp
+var elements = page.Locator("div");
+var divsCount = await elements.EvaluateAll<bool>("(divs, min) => divs.length >= min", 10);
+```
+
+### param: Locator.evaluateAll.expression = %%-evaluate-expression-%%
+
+### param: Locator.evaluateAll.arg
+- `arg` <[EvaluationArgument]>
+
+Optional argument to pass to [`param: expression`].
+
+
+## async method: Locator.evaluateHandle
+- returns: <[JSHandle]>
+
+Returns the return value of [`param: expression`] as a [JSHandle].
+
+This method passes this handle as the first argument to [`param: expression`].
+
+The only difference between [`method: Locator.evaluate`] and [`method: Locator.evaluateHandle`] is that [`method: Locator.evaluateHandle`] returns [JSHandle].
+
+If the function passed to the [`method: Locator.evaluateHandle`] returns a [Promise], then [`method: Locator.evaluateHandle`] would wait
+for the promise to resolve and return its value.
+
+See [`method: Page.evaluateHandle`] for more details.
+
+### param: Locator.evaluateHandle.expression = %%-evaluate-expression-%%
+
+### param: Locator.evaluateHandle.arg
+- `arg` <[EvaluationArgument]>
+
+Optional argument to pass to [`param: expression`].
+
+### option: Locator.evaluateHandle.timeout = %%-input-timeout-%%
+
+## async method: Locator.fill
+
+This method waits for [actionability](./actionability.md) checks, focuses the element, fills it and triggers an `input` event after filling. Note that you can pass an empty string to clear the input field.
+
+If the target element is not an `<input>`, `<textarea>` or `[contenteditable]` element, this method throws an error. However, if the element is inside the `<label>` element that has an associated [control](https://developer.mozilla.org/en-US/docs/Web/API/HTMLLabelElement/control), the control will be filled instead.
+
+To send fine-grained keyboard events, use [`method: Locator.type`].
+
+### param: Locator.fill.value
+- `value` <[string]>
+
+Value to set for the `<input>`, `<textarea>` or `[contenteditable]` element.
+
+### option: Locator.fill.force = %%-input-force-%%
+### option: Locator.fill.noWaitAfter = %%-input-no-wait-after-%%
+### option: Locator.fill.timeout = %%-input-timeout-%%
+
+## async method: Locator.first
+- returns: <[null]|[ElementHandle]>
+
+Resolves given locator to the first matching DOM element. If no elements matching the query are attached, waits for them up to a given timeout.
```

Comment:
Let's note this is not equivalent to `(await locator.all())[0]` due to waiting?

Reply:
I'll also make it return non-null.

Reply:
Leaving as is for now.